easy_logging_test.go

```diff
@@ -0,0 +1,184 @@
+package gosnowflake
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+)
+
+func TestInitializeEasyLoggingOnlyOnceWhenConfigGivenAsAParameter(t *testing.T) {
+	dir := t.TempDir()
+	logLevel := levelError
+	contents := createClientConfigContent(logLevel, dir)
+	configFilePath := createFile(t, "config.json", contents, dir)
+	easyLoggingInitTrials.reset()
+
+	err := openWithClientConfigFile(t, configFilePath)
+
+	assertNilF(t, err, "open config error")
+	assertEqualE(t, toClientConfigLevel(logger.GetLogLevel()), logLevel, "error log level check")
+	assertEqualE(t, easyLoggingInitTrials.configureCounter, 1)
+
+	err = openWithClientConfigFile(t, "")
+	assertNilF(t, err, "open config error")
+	err = openWithClientConfigFile(t, configFilePath)
+	assertNilF(t, err, "open config error")
+	err = openWithClientConfigFile(t, "/another-config.json")
+	assertNilF(t, err, "open config error")
+
+	assertEqualE(t, toClientConfigLevel(logger.GetLogLevel()), logLevel, "error log level check")
+	assertEqualE(t, easyLoggingInitTrials.configureCounter, 1)
+
+	cleanUp()
```

Comment:
It would be better, to use `t.Cleanup` before the assertions? If an assertion fails, cleanup is never executed.